### PR TITLE
Add a platform field to Theme

### PR DESCRIPTION
**Pull request by abarth**
We'll use this field to adapt material widgets to iOS.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/foundation/platform.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The platform that user interaction should adapt to target.
+enum TargetPlatform {
+  /// Android: <https://www.android.com/>
+  android,
+
+  // iOS: <http://www.apple.com/ios/>
```
Comment (eseidelGoogle) [inline packages/flutter/lib/src/foundation/platform.dart]: Did you mean this to be /// ?

Comment (abarth) [inline packages/flutter/lib/src/foundation/platform.dart]: Fixed.

File: packages/flutter/lib/src/foundation/platform.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The platform that user interaction should adapt to target.
+enum TargetPlatform {
```
Comment (HansMuller) [inline packages/flutter/lib/src/foundation/platform.dart]: Do we want suffixes here? Or maybe we're just assuming that the adaptation matches whatever the current version of the platform was when Flutter was built?

File: packages/flutter/lib/src/material/app.dart [hunk 3]
```diff
@@ -25,6 +23,15 @@ const TextStyle _errorTextStyle = const TextStyle(
   decorationStyle: TextDecorationStyle.double
 );
 
+/// The visual and interaction design for overscroll.
+enum OverscrollStyle {
+  /// Overscrolls are clamped and indicated with a glow.
+  glow,
+
+  /// Overscrolls are now clamped and indicated with elastic physics.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app.dart]: We don't clamp the scrollOffset in this case.

File: packages/flutter/lib/src/foundation/platform.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/foundation/platform.dart]: We can always introduce another "version" enum if we want to get more specific in our targetting.

File: packages/flutter/lib/src/material/app.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/lib/src/material/app.dart]: Fixed.

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/material/theme_data.dart [hunk 4]
```diff
@@ -121,6 +125,7 @@ class ThemeData {
     primaryTextTheme ??= primaryIsDark ? Typography.white : Typography.black;
     iconTheme ??= isDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
     primaryIconTheme ??= primaryIsDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
+    platform ??= (Platform.isIOS || Platform.isMacOS) ? TargetPlatform.iOS : TargetPlatform.android;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: this should gets its value from a getter in foundation/platform.dart

Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: and that getter should throw an exception saying that a new enum should be added if it's a new platform

**Pull request merged**